### PR TITLE
pkg/apis: add ability to define template delimiters

### LINKED ISSUE #584: Configurable template delimeters
First of all, thank you so much for this project and for adding https://github.com/bitnami-labs/sealed-secrets/pull/580 in the latest release! This allowed me to remove my dependency on [ConfigMapSecret](https://github.com/machinezone/configmapsecrets) and use only SealedSecrets instead!

I do have one issue with configuration files that contain golang-based templates ([mainly alertmanager configuration like this one](https://github.com/thaum-xyz/ankhmorpork/blob/master/apps/monitoring/manifests/alertmanager/secret.yaml)). Would it be possible to either use non-default delimiters for injecting secret into larger config? Or maybe it would be possible to allow configuring delimiters and allowing something different than the default `{{` and `}}`? 

Just to note, there is an option for this in the standard library: https://golang.org/pkg/text/template/#Template.Delims

### PR DESCRIPTION
**Pull request by paulfantom**
**Description of the change**

Adding ability to configure template delimiters instead of using hardcoded `{{` and `}}`.

**Benefits**

This allows to use templates for SealedSecrets that should contain `{{` and `}}` in resulting Secret.

Delimiters can be configured per SealedSecret resource, so in one environment different delimiters can be used for different SealedSecrets.

**Possible drawbacks**

No drawbacks I can think of.

**Applicable issues**

- fixes #584

**Additional information**

### CODE REVIEW

File: pkg/apis/sealed-secrets/v1alpha1/sealedsecret_expansion.go [hunk 1]
```diff
@@ -285,12 +285,16 @@ func (s *SealedSecret) Unseal(codecs runtimeserializer.CodecFactory, privKeys ma
 
 		for key, value := range s.Spec.Template.Data {
 			var plaintext bytes.Buffer
-			template, err := template.New(key).Parse(value)
+			t := template.New(key)
+			if (s.Spec.Template.Delimeters.Left != "") && (s.Spec.Template.Delimeters.Right != "") {
```
Comment (mkmik) [inline pkg/apis/sealed-secrets/v1alpha1/sealedsecret_expansion.go:289]: If somebody passes this config:

```yaml
spec:
  template:
    delimeters:
      right: "<"
```

then the setting will be silently ignored, which is confusing.

The go template docs says:

> An empty delimiter stands for the corresponding default: {{ or }}

So I think it's safe to just call `t.Delims(s.Spec.Template.Delimeters.Left, s.Spec.Template.Delimeters.Right)` without any conditional

Comment (paulfantom) [reply pkg/apis/sealed-secrets/v1alpha1/sealedsecret_expansion.go:289]: Yeah, I agree. I'll try it just after I figure out how to force `make generate` to work :smile:

Comment (mkmik) [reply pkg/apis/sealed-secrets/v1alpha1/sealedsecret_expansion.go:289]: oh I forgot about 

```
bash: ../../../../vendor/k8s.io/code-generator/generate-groups.sh: No such file or directory
```

**Comment by josvazg** [comment]
We are archiving this PR, it has been open without action for far too long.

We may revisit it from the open issue to decide whether we resume work on this.

@paulfantom feel free to re-post an updated PR if still interested in this feature.

**Pull request closed without merging**